consensus/contracts.go

```diff
@@ -76,78 +71,202 @@ func (s *State) validProof(sp StorageProof) error {
 
 // addContract takes a FileContract and its corresponding ContractID and adds
 // it to the state.
-func (s *State) applyContract(contract FileContract, id ContractID) (cd ContractDiff) {
-	s.openContracts[id] = contract
-	cd = ContractDiff{
-		New:      true,
-		ID:       id,
-		Contract: contract,
+func (s *State) applyFileContracts(bn *blockNode, t Transaction) {
+	for i, fc := range t.FileContracts {
+		// TODO: Sanity check.
+		// Apply the contract.
+		fcid := t.FileContractID(i)
+		s.fileContracts[fcid] = fc
+
+		// Add the diff to the block node.
+		fcd := FileContractDiff{
+			New:          true,
+			ID:           fcid,
+			FileContract: fc,
+		}
+		bn.fileContractDiffs = append(bn.fileContractDiffs, fcd)
 	}
 	return
 }
 
-// applyStorageProof takes a storage proof and adds any outputs created by it
-// to the consensus state.
-func (s *State) applyStorageProof(sp StorageProof) (od OutputDiff, cd ContractDiff) {
-	// Calculate the new output and its id.
-	contract := s.openContracts[sp.ContractID]
-	output := SiacoinOutput{
-		Value:     contract.Payout,
-		SpendHash: contract.ValidProofAddress,
+func (s *State) applyFileContractTerminations(bn *blockNode, t Transaction) {
+	for _, fct := range t.FileContractTerminations {
+		// Delete the contract.
+		fc, exists := s.fileContracts[fct.ParentID]
+		// Sanity check - termination should be terminating an existing
+		// contract.
+		if !exists {
+			if DEBUG {
+				panic("file contract termination terminates a nonexisting contract")
+			} else {
+				return
+			}
```

Comment:
`else` is unnecessary here. I think in general `DEBUG` checks shouldn't have `else` blocks.